operations/src/main/java/com/nextdoor/bender/operation/gelf/GelfOperation.java

```diff
@@ -85,6 +87,49 @@ protected InternalEvent prefix(InternalEvent ievent) {
     return ievent;
   }
 
+  /**
+   * Parses through the internal JSON Deserialized Event and optionally adds a "timestamp" key.
+   *
+   * If the "timestamp" key already exists, this method does nothing and bails out. If there
+   * is no "timestamp" key, and either the EventTime or ArrivalTime of the {@link InternalEvent}
+   * have been set (presumably by the {@link TimeOperation}), then these are used to set the
+   * "timestamp" key in the object.
+   *
+   * @param ievent
+   * @return {@link InternalEvent}
+   */
+  protected InternalEvent setTimestamp(InternalEvent ievent) {
+    DeserializedEvent devent;
+    if ((devent = ievent.getEventObj()) == null) {
+      return null;
+    }
+
+    Object payload = devent.getPayload();
+
+    if (payload == null) {
+      return null;
+    }
+
+    JsonObject obj = (JsonObject) payload;
+
+    if (obj.has("timestamp")) {
+      return ievent;
+    }
+
+    double timestamp;
+    if (ievent.getEventTime() > 1) {
+      timestamp = ievent.getEventTime() / 1000.0;
```

Comment:
By default `eventTime` is set to `arrivalTime` so this if is not needed.

Reply:
Removed the `else if ..` but left the other part of the If statement in there just _in case_.